src/nupic/os/Regex.cpp

```diff
@@ -46,7 +47,14 @@ namespace nupic
       exactRegExp += re;
       if (re[re.length()-1] != '$') 
         exactRegExp += '$';
-      
+
+#if defined(NTA_OS_WINDOWS)
```

Comment:
I really want to minimize platform-specific hacks like this. Did you try to get pcre to work on windows? Or did you check if `std::regex` will work on other platforms?